src/content/lib/utils.js

```diff
@@ -35,30 +36,70 @@ const FAKE_BUGS = [
   },
 ];
 
+function getFakeBugsRequest(bodyString) {
+  return new Request(`/api/bugs?fakebody=${bodyString}`);
+}
+
 export async function runQuery(query) {
   if (prefs.get("offline_debug")) {
     return FAKE_BUGS;
   }
   let data = {};
-  const resp = await fetch("/api/bugs", {
+  let body = JSON.stringify(query);
+  let request = new Request("/api/bugs", {
     headers: {
       Accept: "application/json",
       "Content-Type": "application/json",
     },
-    body: JSON.stringify(query),
+    body,
     method: "POST",
   });
+  let fakeRequest = getFakeBugsRequest(body);
+  let resp = await fetch(request);
+  await cache.set(fakeRequest, resp);
```

Comment:
I don't think we can abstract away the body string part, since the body can only be used once. So something like
```js
  let request = new Request("/api/bugs", {
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body,
    method: "POST",
  });
  let resp = await fetch(request);
  await cache.set(request, resp);
```
won't work. This would:
```js
  let request = new Request("/api/bugs", {
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body,
    method: "POST",
  });
  let resp = await fetch(request);
  await cache.set(body, resp);
```
But I guess it just feels a bit more opaque since it implies that we're keying the response to the body, when in reality we can only key responses to requests.